### PR TITLE
Add receipt printer driver

### PR DESCRIPTION
**Pull request by ppkn**
Adds a simple receipt printer driver. On my system I have these claims in my setup.folk
```tcl
Claim tm-l90 is the default printer
Claim tm-l90 is a receipt printer
Claim tm-l90 is at 192.168.0.156
Claim tm-l90 has tag geometry {tagSize 30mm top 24mm right 28.5mm left 14mm bottom 75.5mm}
```

I tried to use a nested `When` initially, but I wasn't able to do an early return and it was behaving weird if the printer claims were retracted/reclaimed e.g. if they were on a printed program that went missing and came back.

### CODE REVIEW

File: virtual-programs/receipt_printer.folk
Comment (ppkn) [inline virtual-programs/receipt_printer.folk:5]: When printing programs the traditional way, we abort the print if the program already exists. That's not how I remember it but that's the way the code looks. I could add that here but it's nice to be able to reprint a program if the code gets updated.

File: virtual-programs/receipt_printer.folk
Comment (ppkn) [inline virtual-programs/receipt_printer.folk:14]: I'd love to split this template over multiple lines, but I don't know a good way of doing that in Tcl.

Comment (osnr) [reply virtual-programs/receipt_printer.folk]: Maybe make a list and join by ""?

**Review by osnr — APPROVED** [review]
It looks good -- maybe we can merge this into main once checkerboard-calibration is merged. I tested normal printing and it works but obviously didn't get to test receipt printing (although maybe I can at home)

I'm slightly concerned that the logic for making .folk files is now duplicated between places but maybe ok for now

**Comment by ppkn** [comment]
@osnr How do we want to handle previously printed programs? We currently have [code](https://github.com/FolkComputer/folk/blob/ab1541b59896448b2e57af272e0b097ec7f4e843/virtual-programs/print.folk#L313) that throws an error when you try to reprint an existing program.

**Comment by osnr** [comment]
> @osnr How do we want to handle previously printed programs? We currently have [code](https://github.com/FolkComputer/folk/blob/ab1541b59896448b2e57af272e0b097ec7f4e843/virtual-programs/print.folk#L313) that throws an error when you try to reprint an existing program.

Empirically, I think that safeguard is pretty important (it's easy to end up in terrible situations where you accidentally stomp your old programs and your printed programs are inaccurate). I guess the useful application of this is if you want to fully replace an old program, both source and layout? But then you should just delete the files by hand, maybe.

**Comment by ppkn** [comment]
I think the previously printed program safeguard is the last thing I want to resolve before merging this.

>it's easy to end up in terrible situations where you accidentally stomp your old programs and your printed programs are inaccurate

I'm not sure I follow. If you have an existing program open in web-editor, there's nothing that prevents stomping old programs ([code](https://github.com/FolkComputer/folk/blob/ab1541b59896448b2e57af272e0b097ec7f4e843/virtual-programs/web/web-editor.folk#L75)). In which case, I feel like someone would actually want the updated code printed.

Maybe the distinction is that we want to see if the code requested to print is different than what exists on disk?
<dl>
<dt>New program</dt>
<dd>just save and print</dd>
<dt>Code is the same as $id.folk</dt>
<dd>Allow to reprint</dd>
<dt>Code is different from $id.folk</dt>
<dd>??? maybe throw an error and require a save before printing?</dd>
</dl>

File: virtual-programs/receipt-printer.folk
Comment (ppkn) [inline virtual-programs/receipt-printer.folk:34]: @osnr I tried to find a way to `return [uplevel "subst $singleLine"]` here but it kept acting weird so I just call the subst on line 27 🤷 

I guess I need to hone my Tcl skills some more.

Comment (osnr) [reply virtual-programs/receipt-printer.folk:34]: Maybe `[uplevel [list subst $singleLine]]`?

**Pull request merged**
